### ISSUE TITLE
Excessive memory usage and potential non-termination in induction reduction

### ISSUE THREAD

**Issue by gebner** [gapt/gapt#699]
The following code happily consumes all the memory on my machine.
```scala
import at.logic.gapt.examples.theories.nat._
val p = instanceProof(addcomm.combined(), Seq(le"s (s 0)", le"0"))
inductionNormalForm(p)
```

**Comment by jannikvierling** [comment]
Thanks for the bug report. I am already working on this bug as it occurs in the user manual.

**Comment by jannikvierling** [comment]
Fixed by PR #701.
